src/txn/txn_recover.c

```diff
@@ -522,6 +532,15 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
 done:	FLD_SET(conn->log_flags, WT_CONN_LOG_RECOVER_DONE);
 err:	WT_TRET(__recovery_free(&r));
 	__wt_free(session, config);
+
+	/*
+	 * Destroy the eviction threads that were started in support of
+	 * recovery.  They will be restarted once the lookaside table is
+	 * created.
+	 */
+	if (eviction_started)
+		WT_TRET(__wt_evict_destroy(session));
+
```

Comment:
It hides information that's better exposed -- it would never occur to me to look in a function named `__wt_txn_recover` to reason about when eviction is running. When looking at `__wt_connection_workers` to understand the startup code, it's easy to see a start/stop of eviction, but I'm never going to think one of the underlying functions also starts/stops eviction threads.

I worry somebody will shuffle the function calls in `__wt_connection_workers` for some reason, and things will stop working because they don't realize `__wt_txn_recover` isn't just recovery.

Secondarily, it's more complicated to do that work in `__wt_txn_recover` than to wrap that function in calls to start/stop eviction, so it's more error-prone.